docs/testrunner-toolkit/configuration/espresso.md

```diff
@@ -111,6 +111,112 @@ __Example__:
     - "10.0"
 ```
 
+## `devices`
+
+__Description__: Field for defining devices request.
+
+A real device can be selected using two different manner:
+- Specific device selection, through device IDs
+- Filtered device selection, with several contraints such as `name`, `platformVersion`, etc...
+
+When an ID is specified, it is prioritary on other settings.
+
+__Type__: *[]object*
+
+__Example__:
+```yaml
+devices:
+  - name: "Google Pixel.*"
+    platformVersion: 8.1
+  - id: Google_Pixel_2_real_us
+```
+
+### `id`
+
+__Description__: Request a specific device by its id.
+
+__Type__:  *string*
+
+__Example__:
+
+```yaml
+        id: Google_Pixel_2_real_us
+```
+
+### `name`
+
+__Description__:  Request a device by its name.
+
+__Type__:  *string*
+
+__Example__:
+
+Using complete name:
+```yaml
+      - name: Google Pixel 4 XL
+```
+
+Using pattern matching:
+```yaml
+        name: Google Pixel.*
+```
+
+### `platformVersion`
+
+__Description__: Request a device with a specific platform version.
+
+__Type__:  *string*
+
+__Example__:
+
+```yaml
+        platformVersion: 8.1
+```
+
+### `options`
```

Comment:
Technically you can request a device without any constraints (no `id`, no `name` and no `options`) from `saucectl` and `rdc` perspective. I don't know how likely a user will do that as you don't know on which device it will be executed.

As the name is the more important settings, it was at the root level. This have been defined when thinking to config format during a huddle. You can raise this question during today's standup !

Reply:
I'm sorry I didn't see your comment before standup today. So I am curious, could I conceivably request a devices based ONLY on one of the options values? Like, `deviceType: TABLET`? If so, I am not sure why these device request filters are not ALL at the devices root level?

Reply:
@valentyn88 @alexplischke do you remember the reason those items where under `options` ?

Reply:
No particular strong reason either afaik. It was a choice between having it at the device level and under options. No one spoke out against having those fields under `options` and I faintly recall Kunal's preference for it. Simple as that 😄

Reply:
Fair enough.